apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/listener/Reseter.java

```diff
@@ -0,0 +1,157 @@
+ **/
+public enum Reseter {
+    INSTANCE;
+    private static final ILog logger = LogManager.getLogger(Reseter.class);
+    private static final String APPLICATION_ID_NAM = "application_id";
+    private static final String INSTANCE_ID_NAME = "instance_id";
+    private static final String STATUS_NAME = "status";
+    private static final String RESET_CHILD_DIR = "/reset.status";
+    private static final String COMMENT = "#Status has three values: ON (trigger reset), DONE(reset complete), OFF(agent fist boot).\n" +
+        "Application_id: application_id of the current agent.\n" +
+        "Instance_id: the instanceid of the current agent.";
+    private volatile Properties properties = new Properties();
+    private String resetPath;
+    private ResetStatus status = ResetStatus.OFF;
+    private int isFirstRun = 0;
+    private int detectDuration = 5;
+
+    public Reseter setStatus(ResetStatus status) {
+        this.status = status;
+        return this;
+    }
+
+    public String getResetPath() throws IOException {
+        if (isFirstRun == 0) {
+            File statusDir = new File(Config.Agent.REGISTER_STATUS_DIR);
+            if (!statusDir.exists() || !statusDir.isDirectory()) {
+                statusDir.mkdir();
+            }
+            resetPath = statusDir.getAbsolutePath() + RESET_CHILD_DIR;
+            init();
+            isFirstRun = 1;
+        }
+        return resetPath;
+    }
+
+    public void reportToRegisterFile() throws IOException {
+        FileOutputStream outputStream = null;
+        try {
+            File configFile = new File(resetPath);
+            properties.setProperty(APPLICATION_ID_NAM, RemoteDownstreamConfig.Agent.SERVICE_ID + "");
+            properties.setProperty(INSTANCE_ID_NAME, RemoteDownstreamConfig.Agent.SERVICE_INSTANCE_ID + "");
+            properties.setProperty(STATUS_NAME, status.value());
+            outputStream = new FileOutputStream(configFile);
+            properties.store(outputStream, COMMENT);
+        } finally {
+            closeFileStream(outputStream);
+        }
+    }
+
+    public Reseter clearID() {
+        RemoteDownstreamConfig.Agent.SERVICE_ID = DictionaryUtil.nullValue();
+        RemoteDownstreamConfig.Agent.SERVICE_INSTANCE_ID = DictionaryUtil.nullValue();
+        EndpointNameDictionary.INSTANCE.clearEndpointNameDictionary();
+        NetworkAddressDictionary.INSTANCE.clearNetworkAddressDictionary();
+        ServiceManager.INSTANCE.findService(TraceSegmentServiceClient.class).clearCache();
+        status = ResetStatus.DONE;
+        logger.info("clear id successfully,begin trigger reset.");
+        return this;
+    }
+
+    public Boolean predicateReset() throws IOException {
+        File resetFile = new File(getResetPath());
+        FileInputStream inputStream = null;
+        FileLock fileLock = null;
+        FileChannel fileChannel = null;
+        if (System.currentTimeMillis() - resetFile.lastModified() < detectDuration * 1000) {
```

Comment:
If `detectDuration` is set to 5 only, why multiply 1000 here? Suggest set to 5000.